src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -52,6 +56,12 @@ public EnumConverter(EnumConverterOptions converterOptions, JsonSerializerOption
 
         public EnumConverter(EnumConverterOptions converterOptions, JsonNamingPolicy? namingPolicy, JsonSerializerOptions serializerOptions)
         {
+            // If enum contains special char, make it failed to serialize or deserialize.
+            if (s_containsSpecialChar)
+            {
+                ThrowHelper.ThrowJsonException();
```

Comment:
@eiriktsarpalis  Do you think we can add meaningful message in exception?

Reply:
Converters failing to initialize typically throw `InvalidOperationException`, e.g.

https://github.com/dotnet/runtime/blob/10fc8ae9a2af0a0ced69de92bb0d7734ff35946b/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectConverterFactory.cs#L50-L53

Yes, it would require a special error message, which can be added using a new entry in `String.resx` and `ThrowHelpers.cs`.